packages/datadog-plugin-mongodb-core/test/mongodb.spec.js

```diff
@@ -108,11 +108,16 @@ describe('Plugin', () => {
             }, () => {})
           })
 
-          it('should sanitize the query', done => {
+          it('should sanitize the query', function (done) {
+            this.timeout(1000 * 60)
+            const queryObjectDepth = 5500
```

Comment:
Why go that deep? Is this something we actually want to support?

Reply:
at 5000 the test passes with the old implementation, I was just stepping up by 500 until it crashed. in all likelyhood hitting this limit is somewhat unreasonable in real world situations. perhaps the more realistic situation is encountering a cyclic object that causes this.

Reply:
Can we put an upper bound instead and test that it can't go beyond it? You mentioned 20 which I think would be reasonable.

Reply:
done, and rebased to keep commits back down